zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -329,7 +329,7 @@ cdef _array_for_column_impl(object dtype,
                 # upper bound doesn't include the timestamp because we've
                 # already included the timestamp-date in the baseline.
                 end = max(ts_ix - 1, 0)
-                if end >= asof_ix:
+                if end >= asof_ix and out_array[asof_ix, column_ix] != value:
```

Comment:
Can we add a comment here explaining why we don't need to emit an adjustment if the new value equals the already-written value?